contracts/ERC721A.sol

```diff
@@ -575,6 +573,15 @@ contract ERC721A is IERC721A {
         _afterTokenTransfers(from, to, tokenId, 1);
     }
 
+    /**
+     * @dev Packs ownership data into single uint
+     */
+    function _packTransferOwnershipData(address to) private view returns (uint256 packedData) {
+        assembly {
+            packedData := or(to, or(shl(BITPOS_START_TIMESTAMP, timestamp()), BITMASK_NEXT_INITIALIZED))
+        }
+    }
+
```

Comment:
Refactor it into:

```solidity
/**
 * @dev Packs ownership data.
 */
function _packOwnershipData(address owner, uint256 flags) private view returns (uint256 value) {
    assembly {
        value := or(owner, or(shl(BITPOS_START_TIMESTAMP, timestamp()), flags))
    }
}
```

And use it across `_mint`, `_transfer`, `_burn`.

Should give similar results.

Reply:
Somehow the compiler is so stubborn for this case that it needs some assembly to whip it into place.